Source/Core/VideoBackends/Metal/MTLUtil.mm

```diff
@@ -0,0 +1,480 @@
+  if ([[device name] containsString:@"NVIDIA"])
+    vendor = DriverDetails::VENDOR_NVIDIA;
+  else if ([[device name] containsString:@"AMD"])
+    vendor = DriverDetails::VENDOR_ATI;
+  else if ([[device name] containsString:@"Intel"])
+    vendor = DriverDetails::VENDOR_INTEL;
+  else if ([[device name] containsString:@"Apple"])
+    vendor = DriverDetails::VENDOR_APPLE;
+  double version = 0;
+  if (@available(macOS 13, iOS 16, *))
+    version = 3.0;
+  else if (@available(macOS 12, iOS 15, *))
+    version = 2.4;
+  else if (@available(macOS 11, iOS 14, *))
+    version = 2.3;
+  else if (@available(macOS 10.15, iOS 13, *))
+    version = 2.2;
+  else if (@available(macOS 10.14, iOS 12, *))
+    version = 2.1;
+  else if (@available(macOS 10.13, iOS 11, *))
+    version = 2.0;
+  DriverDetails::Init(DriverDetails::API_METAL, vendor, DriverDetails::DRIVER_APPLE, version,
+                      DriverDetails::Family::UNKNOWN);
+
+#ifdef TARGET_OS_MAC
+  config->backend_info.bSupportsDepthClamp = true;
+  config->backend_info.bSupportsST3CTextures = true;
+  config->backend_info.bSupportsBPTCTextures = true;
+#else
+  bool supports_mac1 = false;
+  bool supports_apple4 = false;
+  if (@available(iOS 13, *))
+  {
+    supports_mac1 = [device supportsFamily:MTLGPUFamilyMac1];
+    supports_apple4 = [device supportsFamily:MTLGPUFamilyApple4];
+  }
+  else
+  {
+    supports_apple4 = [device supportsFeatureSet:MTLFeatureSet_iOS_GPUFamily4_v1];
+  }
+  config->backend_info.bSupportsDepthClamp = supports_mac1 || supports_apple4;
+  config->backend_info.bSupportsST3CTextures = supports_mac1;
+  config->backend_info.bSupportsBPTCTextures = supports_mac1;
+  config->backend_info.bSupportsFramebufferFetch = true;
+#endif
+
+  config->backend_info.AAModes.clear();
+  for (u32 i = 1; i <= 64; i <<= 1)
+  {
+    if ([device supportsTextureSampleCount:i])
+      config->backend_info.AAModes.push_back(i);
+  }
+
+  // The unified memory path (using shared buffers for everything) performs noticeably better with
+  // bbox even on discrete GPUs (20fps vs 15fps in Super Paper Mario elevator), so default to that.
+  // The separate buffer + manual upload path is left available for testing and comparison.
+  if (char* env = getenv("MTL_UNIFIED_MEMORY"))
+    g_features.unified_memory = env[0] == '1' || env[0] == 'y' || env[0] == 'Y';
+  else
+    g_features.unified_memory = true;
+
+  g_features.subgroup_ops = false;
+  if (@available(macOS 10.15, iOS 13, *))
+  {
+    // Requires SIMD-scoped reduction operations
+    g_features.subgroup_ops =
+        [device supportsFamily:MTLGPUFamilyMac2] || [device supportsFamily:MTLGPUFamilyApple6];
+    config->backend_info.bSupportsFramebufferFetch = [device supportsFamily:MTLGPUFamilyApple1];
+  }
+  if (DriverDetails::HasBug(DriverDetails::BUG_BROKEN_SUBGROUP_INVOCATION_ID))
+    g_features.subgroup_ops = false;
+#ifdef TARGET_OS_MAC
+  if (version >= 2.3 && vendor == DriverDetails::VENDOR_INTEL)
```

Comment:
Is it possible to replace the version check with an `@available`?

Reply:
Can't use `&&` with `@available` (or at least not without compiler warnings) so it'll have to be `if (@available(...)) if (vendor == DriverDetails::VENDOR_INTEL)`.  Is that fine?

Reply:
Seems fine to me.